From dd8e09b01eda5b9b766a379b5c0722c78d8dd645 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Thu, 11 Oct 2012 22:38:26 +0200
Subject: [PATCH] Revert "timedated: replace ntp-units file with an
 ntp-units.d drop-in dir"

This reverts commit 7ec9c50ae6e3329926dcae48c661910043701f5a.
---
 Makefile.am              |  4 ++++
 src/timedate/ntp-units   |  4 ++++
 src/timedate/timedated.c | 59 ++++++++++++++++++------------------------------
 3 files changed, 30 insertions(+), 37 deletions(-)
 create mode 100644 src/timedate/ntp-units

diff --git a/Makefile.am b/Makefile.am
index d051b39..46cfe80 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -114,6 +114,7 @@ AM_CPPFLAGS = \
 	-DSYSTEM_SHUTDOWN_PATH=\"$(systemshutdowndir)\" \
 	-DSYSTEM_SLEEP_PATH=\"$(systemsleepdir)\" \
 	-DSYSTEMD_KBD_MODEL_MAP=\"$(pkgdatadir)/kbd-model-map\" \
+	-DSYSTEMD_NTP_UNITS=\"$(pkgdatadir)/ntp-units\" \
 	-DX_SERVER=\"$(bindir)/X\" \
 	-DPOLKIT_AGENT_BINARY_PATH=\"$(bindir)/pkttyagent\" \
 	-I $(top_srcdir)/src \
@@ -2061,6 +2062,9 @@ timedated-install-data-hook:
 INSTALL_DATA_HOOKS += \
 	timedated-install-data-hook
 
+dist_pkgdata_DATA += \
+	src/timedate/ntp-units
+
 EXTRA_DIST += \
 	units/systemd-timedated.service.in
 endif
diff --git a/src/timedate/ntp-units b/src/timedate/ntp-units
new file mode 100644
index 0000000..2a46f66
--- /dev/null
+++ b/src/timedate/ntp-units
@@ -0,0 +1,4 @@
+# NTP service implementations, in order of preference
+
+chronyd.service
+ntpd.service
diff --git a/src/timedate/timedated.c b/src/timedate/timedated.c
index 1dcf509..8be45d3 100644
--- a/src/timedate/timedated.c
+++ b/src/timedate/timedated.c
@@ -31,7 +31,6 @@
 #include "polkit.h"
 #include "def.h"
 #include "hwclock.h"
-#include "conf-files.h"
 
 #define NULL_ADJTIME_UTC "0.0 0 0\n0\nUTC\n"
 #define NULL_ADJTIME_LOCAL "0.0 0 0\n0\nLOCAL\n"
@@ -305,54 +304,40 @@ static int write_data_local_rtc(void) {
 }
 
 static char** get_ntp_services(void) {
-        char **r = NULL, **files, **i;
-        int k;
-
-        k = conf_files_list(&files, ".list",
-                            "/etc/systemd/ntp-units.d",
-                            "/run/systemd/ntp-units.d",
-                            "/usr/local/lib/systemd/ntp-units.d",
-                            "/usr/lib/systemd/ntp-units.d",
-                            NULL);
-        if (k < 0)
-                return NULL;
-
-        STRV_FOREACH(i, files) {
-                FILE *f;
+        char **r = NULL;
+        FILE *f;
 
-                f = fopen(*i, "re");
-                if (!f)
-                        continue;
+        f = fopen(SYSTEMD_NTP_UNITS, "re");
+        if (!f)
+                return NULL;
 
-                for (;;) {
-                        char line[PATH_MAX], *l, **q;
+        for (;;) {
+                char line[PATH_MAX], *l, **q;
 
-                        if (!fgets(line, sizeof(line), f)) {
+                if (!fgets(line, sizeof(line), f)) {
 
-                                if (ferror(f))
-                                        log_error("Failed to read NTP units file: %m");
+                        if (ferror(f))
+                                log_error("Failed to read NTP units file: %m");
 
-                                break;
-                        }
+                        break;
+                }
 
-                        l = strstrip(line);
-                        if (l[0] == 0 || l[0] == '#')
-                                continue;
+                l = strstrip(line);
+                if (l[0] == 0 || l[0] == '#')
+                        continue;
 
-                        q = strv_append(r, l);
-                        if (!q) {
-                                log_error("Out of memory");
-                                break;
-                        }
 
-                        strv_free(r);
-                        r = q;
+                q = strv_append(r, l);
+                if (!q) {
+                        log_error("Out of memory");
+                        break;
                 }
 
-                fclose(f);
+                strv_free(r);
+                r = q;
         }
 
-        strv_free(files);
+        fclose(f);
 
         return r;
 }
